src/main/java/com/kichik/pecoff4j/ModifiableStructure.java

```diff
@@ -0,0 +1,21 @@
+package com.kichik.pecoff4j;
+
+import com.kichik.pecoff4j.io.IDataWriter;
+
+import java.io.IOException;
+
+public interface ModifiableStructure {
+    /**
+     * Update computed values such as lengths, checksums, serialized layout etc.
+     *
+     * @return the updated length of this structure in bytes (might include padding)
+     */
+    int rebuild();
```

Comment:
Do any of the implementations actually rebuild anything? At least in this diff it seems to only calculate the length and not modify the structure itself.

Reply:
You are right, in the current PR, there are only length calculations. The next PR will tackle the resources tree, where this method will layout the directories and entries, and updating all pointers between them.